test/unit/cli-domain-package-server-script.js

```diff
@@ -6,283 +6,114 @@ var path = require('path');
-
-        packageServerScript({
-          componentPath: '/path/to/component/',
-          ocOptions: {
-            files: {
-              data: 'myserver.js'
-            }
-          },
-          publishPath: '/path/to/component/_package/'
-        }, function(e, r){
-          error = e;
-          done();
-        });
-      });
-
-      it('should throw an error with error details', function(){
-        expect(error.toString()).to.equal('Error: Javascript error found in myserver.js [3,19]: SyntaxError: Unexpected token punc «;», expected punc «,»]');
-      });
-    });
-
-    describe('when component does not require any json', function(){
-
-      var result,
-          serverjs = 'module.exports.data=function(context,cb){return cb(null, {name:\'John\'}); };';
-
-      beforeEach(function(done){
-
-        initialise({ readFileSync: sinon.stub().returns(serverjs) });
-
-        packageServerScript({
-          componentPath: '/path/to/component/',
-          ocOptions: {
-            files: {
-              data: 'server.js'
-            }
-          },
-          publishPath: '/path/to/component/_package/'
-        }, function(e, r){
-          result = r;
-          done();
-        });
-      });
+  describe('bundle/config/externalDependenciesHandlers when configured with a dependencies hash', function(){
+    var handler = externalDependenciesHandlers({lodash: '4.17.14'});
 
-      it('should save compiled data provider', function(){
-        expect(fsMock.writeFile.args[0][1]).to.equal('module.exports.data=function(n,e){return e(null,{name:"John"})};');
-      });
-
-      it('should return hash for script', function(){
-        expect(result.hashKey).not.be.empty;
-      });
+    it('should return an array containing a function and a regular expression ', function(){
+      expect(handler).to.be.an('array');
+      expect(handler.length).to.be.equal(2);
+      expect(handler[1] instanceof RegExp).to.be.true;
+      expect(handler[0]).to.be.a('function');
     });
 
-    describe('when component requires a json', function(){
-
-      var requiredJson = { hello: 'world' },
-          serverjs = 'var data = require(\'./someJson\'); module.exports.data=function(context,cb){return cb(null,{}); };';
-
-      beforeEach(function(done){
-
-        initialise({
-          readFileSync: sinon.stub().returns(serverjs),
-          readJsonSync: sinon.stub().returns(requiredJson)
-        });
-
-        packageServerScript({
-          componentPath: '/path/to/component/',
-          ocOptions: {
-            files: {
-              data: 'server.js'
-            }
-          },
-          publishPath: '/path/to/component/_package/'
-        }, done);
+    describe('its regular expression', function(){
+      var regex = handler[1];
+      it('should match npm module names', function() {
+        expect(regex.test('lodash')).to.be.true;
+        expect(regex.test('lodash/fp/curryN')).to.be.true;
```

Comment:
@matthewdavidson that helped me fine an exception to our regularExpression for the webpack `externals`: 
A relative path from an npm module can in fact contain upperCase letters